### PR TITLE
[SR SQL Planner Summer Camp]Refact admin repair table

### LINKED ISSUE #7945: New Parser and Analyzer support admin related-statements (Easy)
# Required work and output

1.  Use antlr4 to complete parse statements. The generated statement must follow the original data structure, and should be compatible with the old statement.
    
2.  In the original syntax file  **sql_parser.cup**, the new syntax needs to be compatible with the syntax of the old version, using the syntax file sql_parser.cup of the old version as the reference standard.
    
3.  Remove the logic of analytic functions in the statement and move the relevant logic to the new Analyzer framework. The new Analyzer is in the SQL/Analyzer directory.
    
4.  If the code related to Privilege Check exists in the original analyzer, move the code related to privilege control into PrivilegeChecker.
    
5.  When submitting the PR, please make sure there is unit test coverage for related changes.
    

# Tasks

 - [ ] AdminCancelRepairTableStmt
 - [ ] AdminCheckTabletsStmt
 - [ ] AdminRepairTableStmt

# How to participate?

1.  Pick an issue you are interested in, click the issue link and leave a comment under the issue to sign up.  **One person can only choose one issue and the first person who picks the issue will be assigned.**
    
2.  **After the issue is assigned, you should complete the task in 14 days.** If you are unable to complete it within the specified time, the task will be withdrawn and assigned to others.
    
3.  Participants are required to produce two pieces of content (forms including but not limited to mind maps, study notes, blogs, and tutorials) in the course of the activity, each of which should be at least 200 words.
    
4.  Got questions? Join our  [Slack channel](https://join.slack.com/t/starrocks/shared_invite/zt-192zeqlc7-9MAvqAXNcveW_l5Pqr8r2g)  and find channel:  **#sql-planner**.
    

# Before you start

1.  Read  [Contributing.md](https://github.com/StarRocks/starrocks/blob/main/CONTRIBUTING.md)  carefully to understand how to submit a pull request and how to configure BE and FE development environment.
    
2.  Sign the  [Contributor License Agreement](https://cla-assistant.io/StarRocks/starrocks)  before you make a pull request.
    
3.  To learn more about StarRocks, please refer to  [StarRocks documentation](https://github.com/StarRocks/docs/blob/main/TOC.md).
    

# Pull request requirements

1.  Create a pull request (PR) after you complete the task.
* PR title format: [SR SQL Planner Summer Camp] + PR title
2. After submitting PR, you can comment under the pinned Issue [SR SQL Planner Summer Camp].
* Comment format: "I have finished Issue #" + PR link

### PR DESCRIPTION
**Pull request by amorynan**
## What type of PR is this：
- [ ] bug
- [ ] feature
- [ ] enhancement
- [x] refactor
- [ ] others

## Which issues of this PR fixes ：

Fixes #7945 

## Problem Summary(Required) ：

support admin repair table and so on statement
## Checklist:

- [x] I have added test cases for my bug fix or my new feature
- [x] I have added user document for my new feature or new function

### CODE REVIEW

**Comment by amorynan** [comment]
run starrocks_fe_unittest

**Comment by amorynan** [comment]
@imay  @HangyuanLiu  please review to this pr

**Comment by amorynan** [comment]
starrocks_fe_unittest

**Comment by amorynan** [comment]
run starrocks_fe_unittest

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]

:heart_eyes: **pass** : 91 / 100 (91.00%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/sql/ast/AstVisitor.java|0|3|00.00%|[209, 212, 215]|
|:large_blue_circle:|com/starrocks/sql/analyzer/PrivilegeChecker.java|6|9|66.67%|[270, 277, 284]|
|:large_blue_circle:|com/starrocks/sql/analyzer/AdminStmtAnalyzer.java|33|36|91.67%|[148, 171, 191]|
|:large_blue_circle:|com/starrocks/analysis/AdminRepairTableStmt.java|9|9|100.00%|[]|
|:large_blue_circle:|com/starrocks/sql/analyzer/Analyzer.java|6|6|100.00%|[]|
|:large_blue_circle:|com/starrocks/sql/parser/AstBuilder.java|25|25|100.00%|[]|
|:large_blue_circle:|com/starrocks/analysis/AdminCancelRepairTableStmt.java|7|7|100.00%|[]|
|:large_blue_circle:|com/starrocks/analysis/AdminCheckTabletsStmt.java|5|5|100.00%|[]|

File: docs/release_notes/release-2.3.md [hunk 1]
```diff
@@ -2,6 +2,8 @@
 
 ## 2.3.0
 
+Release date: July 29, 2022
```
Comment (HangyuanLiu) [inline docs/release_notes/release-2.3.md:5]:
```suggestion
```

File: docs/release_notes/release-2.3.md [hunk 2]
```diff
@@ -67,3 +69,4 @@ Fix the following bugs:
 - StarGo, a cluster management tool, can deploy, start, upgrade, and roll back clusters and manage multiple clusters. For more information, see [Deploy StarRocks with StarGo](../administration/stargo.md).
 
 - The pipeline engine is enabled by default when you upgrade StarRocks to version 2.3 or deploy StarRocks. The pipeline engine can improve the performance of simple queries in high concurrency scenarios and complex queries. If you detect significant performance regressions when using StarRocks 2.3, you can disable the pipeline engine by executing the `SET GLOBAL` statement to set `enable_pipeline_engine` to `false`.
+- The [SHOW GRANTS](../sql-reference/sql-statements/account-management/SHOW%20GRANTS.md) statement is compatible with the MySQL syntax and displays the privileges assigned to a user in the form of GRANT statements.
```
Comment (HangyuanLiu) [inline docs/release_notes/release-2.3.md:72]:
```suggestion
```

File: fe/fe-core/src/main/java/com/starrocks/sql/parser/AstBuilder.java [hunk 3]
```diff
@@ -2504,12 +2507,12 @@ private DataDescription getDataDescription(StarRocksParser.DataDescContext conte
         List<String> colList = null;
         if (context.colList != null) {
             List<Identifier> identifiers = visit(context.colList.identifier(), Identifier.class);
-            colList = identifiers.stream().map(Identifier::getValue).map(String::toLowerCase).collect(toList());
+            colList = identifiers.stream().map(Identifier::getValue).collect(toList());
```
Comment (HangyuanLiu) [inline fe/fe-core/src/main/java/com/starrocks/sql/parser/AstBuilder.java:2510]: Why do this modification alone, is there a bug here?

Comment (amorynan) [reply fe/fe-core/src/main/java/com/starrocks/sql/parser/AstBuilder.java:2510]: this is another modification which is bug which was fixed and merged 4 days ago ,  when I made this pr , some conflicts resolution staff to modification these lines. I will try to make it was to be

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]

:heart_eyes: **pass** : 89 / 98 (90.82%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/sql/ast/AstVisitor.java|0|3|00.00%|[209, 212, 215]|
|:large_blue_circle:|com/starrocks/sql/analyzer/PrivilegeChecker.java|6|9|66.67%|[270, 277, 284]|
|:large_blue_circle:|com/starrocks/sql/analyzer/AdminStmtAnalyzer.java|33|36|91.67%|[148, 171, 191]|
|:large_blue_circle:|com/starrocks/analysis/AdminRepairTableStmt.java|9|9|100.00%|[]|
|:large_blue_circle:|com/starrocks/sql/analyzer/Analyzer.java|6|6|100.00%|[]|
|:large_blue_circle:|com/starrocks/sql/parser/AstBuilder.java|23|23|100.00%|[]|
|:large_blue_circle:|com/starrocks/analysis/AdminCancelRepairTableStmt.java|7|7|100.00%|[]|
|:large_blue_circle:|com/starrocks/analysis/AdminCheckTabletsStmt.java|5|5|100.00%|[]|

File: docs/release_notes/release-2.3.md [hunk 4]
```diff
@@ -2,8 +2,6 @@
 
 ## 2.3.0
 
-Release date: July 29, 2022
```
Comment (HangyuanLiu) [inline docs/release_notes/release-2.3.md:5]: Please be careful not to modify irrelevant files, please correct rebase code

**Review by HangyuanLiu — APPROVED** [review]

**Comment by HangyuanLiu** [comment]
run starrocks_fe_unittest

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]

:heart_eyes: **pass** : 89 / 98 (90.82%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/sql/ast/AstVisitor.java|0|3|00.00%|[212, 215, 218]|
|:large_blue_circle:|com/starrocks/sql/analyzer/PrivilegeChecker.java|6|9|66.67%|[271, 278, 285]|
|:large_blue_circle:|com/starrocks/sql/analyzer/AdminStmtAnalyzer.java|33|36|91.67%|[158, 181, 201]|
|:large_blue_circle:|com/starrocks/analysis/AdminRepairTableStmt.java|9|9|100.00%|[]|
|:large_blue_circle:|com/starrocks/sql/analyzer/Analyzer.java|6|6|100.00%|[]|
|:large_blue_circle:|com/starrocks/sql/parser/AstBuilder.java|23|23|100.00%|[]|
|:large_blue_circle:|com/starrocks/analysis/AdminCancelRepairTableStmt.java|7|7|100.00%|[]|
|:large_blue_circle:|com/starrocks/analysis/AdminCheckTabletsStmt.java|5|5|100.00%|[]|

**Comment by amorynan** [comment]
@imay  please review this pr , thx.

**Review by kangkaisen — APPROVED** [review]

**Comment by wanpengfei-git** [comment]
run starrocks_admit_test

**Pull request merged**